caffe2/CMakeLists.txt

```diff
@@ -1683,6 +1683,11 @@ if(BUILD_TEST)
       get_filename_component(test_name ${test_src} NAME_WE)
       add_executable(${test_name} "${test_src}")
       target_link_libraries(${test_name} torch_library gtest_main)
+      if(APPLE)
```

Comment:
Huh, when you USE_MPS=1 but APPLE=0 ?
Or is this to single out mobile or something?

Reply:
This is to check if MacOS SDL libs such as Metal or Foundation are present. I'm not sure if it's ever possible to have  USE_MPS=1 on a non-Apple system, so I just wanted to be safe. 
If you think if whenever USE_MPS=1 then it's "always" an Apple OS, then I'll remove that condition and merge the  `target_link_libraries()`

Reply:
That feels like a fair assumption in my head yes.
We can add a small check above to make sure it is true once and for all as well if we're worried.

Reply:
Merged the two `target_link_libraries` and updated the PR.